src/RequestHandle.cpp

```diff
@@ -65,11 +84,37 @@ auto RequestHandle::init() -> void
     curl_easy_setopt(m_curl_handle, CURLOPT_PRIVATE,        this);
     curl_easy_setopt(m_curl_handle, CURLOPT_HEADERFUNCTION, curl_write_header);
     curl_easy_setopt(m_curl_handle, CURLOPT_HEADERDATA,     this);
-    curl_easy_setopt(m_curl_handle, CURLOPT_WRITEFUNCTION,  curl_write_data);
     curl_easy_setopt(m_curl_handle, CURLOPT_WRITEDATA,      this);
     curl_easy_setopt(m_curl_handle, CURLOPT_NOSIGNAL,       1l);
     curl_easy_setopt(m_curl_handle, CURLOPT_FOLLOWLOCATION, 1l);
 
+    /**
+     * If m_max_bytes is set, then we need to set the XFERINFOFUNCTION and use
+     * the partial write function. Otherwise, use the normal write callback.
+     *
+     * Older versions of cURL don't have XFERINFOFUNCTION, so PROGRESSFUNCTION
+     * must be used in those cases. For versions that do support it, XFERINFOFUNCTION
+     * will be preferentially used when both are set.
+     * https://curl.haxx.se/libcurl/c/progressfunc.html
+     **/
+    if (m_max_download_bytes > -1)
```

Comment:
SetMaxDownloadBytes(max_download_bytes) // done, the max download bytes sets all the appropriate curl_easy_setopt() fields correctly and only once

Reply:
Got it! That makes a lot more sense now.